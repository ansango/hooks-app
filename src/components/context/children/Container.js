
import "./Container.css";
export const Container = ({ title = "", children }) => {
  return (
    <div className="base-container py-5 bg-light">
      <h3 className="text-center">{title}</h3>
      <div className="px-2">{children}</div>
    </div>
  );
};
