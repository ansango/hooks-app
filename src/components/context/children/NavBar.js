import { NavLink } from "react-router-dom";

export const NavBar = ({ routes }) => {
  return (
    <nav className="navbar navbar-light bg-light d-flex flex-row justify-content-end">
      <div className="navbar-nav">
        <div className="d-flex">
          {routes.map((route) => (
            <NavLink
              key={route.name}
              exact
              activeClassName="active"
              className="nav-item nav-link mx-3"
              to={route.path}
            >
              {route.name}
            </NavLink>
          ))}
        </div>
      </div>
    </nav>
  );
};
