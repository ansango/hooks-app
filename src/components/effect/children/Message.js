import { useEffect, useState } from "react";

export const Message = () => {
  const [coords, setCoords] = useState({ x: 0, y: 0 });
  const { x, y } = coords;
  useEffect(() => {
    const mouseMove = (e) => setCoords({ x: e.x, y: e.y });
    window.addEventListener("mousemove", mouseMove);
    return () => {
      window.removeEventListener("mousemove", mouseMove);
    };
  }, []);
  return (
    <div>
      X: <p>{x}</p>
      Y: <p>{y}</p>
    </div>
  );
};
