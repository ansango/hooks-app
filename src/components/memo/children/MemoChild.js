import React from "react";

export const MemoChild = React.memo(({ number, increment }) => {
  console.log("render");

  return (
    <button className="btn btn-primary me-3" onClick={() => increment(number)}>
      {number}
    </button>
  );
});
