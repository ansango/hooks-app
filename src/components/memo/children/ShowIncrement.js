import React from "react";
import { log } from "../../../helpers/log";

export const ShowIncrement = React.memo(({ increment }) => {
  log("ShowIncrement", "Render");
  return (
    <button className="btn btn-primary" onClick={() => increment(5)}>
      Incrementar
    </button>
  );
});
