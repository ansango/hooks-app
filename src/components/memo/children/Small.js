import React from "react";
import { log } from "../../../helpers/log";

export const Small = React.memo(({ value }) => {
  log("Small", "Render");
  return <small>{value}</small>;
});
