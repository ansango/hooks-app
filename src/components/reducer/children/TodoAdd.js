import { generateId } from "../../../helpers/generateId";
import { useForm } from "../../../hooks/useForm";

export const TodoAdd = ({ handleAddTodo }) => {
  const [{ description }, handleInputChange, reset] = useForm({
    description: "",
  });

  const handleSubmit = (e) => {
    e.preventDefault();
    if (description.trim().length !== 0) {
      const newTodo = {
        id: generateId(),
        description: description,
        done: false,
      };

      handleAddTodo(newTodo);
      reset();
    }
  };
  return (
    <>
      <h3> Add Todo</h3>
      <hr />
      <form onSubmit={handleSubmit} className="py-2">
        <input
          type="text"
          placeholder="Add a todo"
          name="description"
          autoComplete="off"
          className="form-control mb-3"
          value={description}
          onChange={handleInputChange}
        />
        <button className="btn btn-outline-primary w-100" type="submit">
          Add
        </button>
      </form>
    </>
  );
};
