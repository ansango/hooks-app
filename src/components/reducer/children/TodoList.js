import { TodoListItem } from "./TodoListItem";

export const TodoList = ({ todos, handleDeleteTodo, handleToggleTodo }) => {
  return (
    <ul className="list-group list-group-flush">
      {todos.map((todo, index) => (
        <TodoListItem
          key={todo.id}
          todo={todo}
          index={index}
          handleDeleteTodo={handleDeleteTodo}
          handleToggleTodo={handleToggleTodo}
        />
      ))}
    </ul>
  );
};
