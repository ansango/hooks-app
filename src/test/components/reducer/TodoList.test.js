import { shallow } from "enzyme";
import { TodoList } from "../../../components/reducer/children/TodoList";
import { todos } from "../../fixtures/todos";
describe("Test in <TodoList/>", () => {
  const handleToggleTodo = jest.fn();
  const handleDeleteTodo = jest.fn();

  const wrapper = shallow(
    <TodoList
      todos={todos}
      handleDeleteTodo={handleDeleteTodo}
      handleToggleTodo={handleToggleTodo}
    />
  );
  test("should render component", () => {
    expect(wrapper).toMatchSnapshot();
  });

  test("should has three <TodoListItem/> components", () => {
    expect(wrapper.find("TodoListItem").length).toBe(todos.length);
    expect(wrapper.find("TodoListItem").at(0).prop("handleDeleteTodo")).toEqual(
      expect.any(Function)
    );
  });
});
