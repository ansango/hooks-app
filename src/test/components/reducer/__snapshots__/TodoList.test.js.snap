// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test in <TodoList/> should render component 1`] = `
<ul
  className="list-group list-group-flush"
>
  <TodoListItem
    handleDeleteTodo={[MockFunction]}
    handleToggleTodo={[MockFunction]}
    index={0}
    key="1"
    todo={
      Object {
        "description": "Learn React",
        "done": false,
        "id": 1,
      }
    }
  />
  <TodoListItem
    handleDeleteTodo={[MockFunction]}
    handleToggleTodo={[MockFunction]}
    index={1}
    key="2"
    todo={
      Object {
        "description": "Learn Mongo",
        "done": false,
        "id": 2,
      }
    }
  />
  <TodoListItem
    handleDeleteTodo={[MockFunction]}
    handleToggleTodo={[MockFunction]}
    index={2}
    key="3"
    todo={
      Object {
        "description": "Learn Node",
        "done": false,
        "id": 3,
      }
    }
  />
</ul>
`;
